src/main/java/com/eprosima/fastcdr/idl/templates/TypesSourcev1.stg

```diff
@@ -0,0 +1,2469 @@
+    $preffix$$name$ = ($typecode.scopedname$)enum_value;
+}
+$elseif(typecode.isBitmaskType)$
+{
+    $typecode.castingType$ bitmask_value = 0;
+    dcdr \>> bitmask_value;
+    $preffix$$name$ = ($typecode.scopedname$)bitmask_value;
+}
+
+$elseif(ctx.generateTypesC)$
+$if(typecode.isStringType)$
+{
+    std::string aux;
+    dcdr \>> aux;
+    free($preffix$$name$);
+    $preffix$$name$ = (char*)malloc(aux.size() + 1);
+    strncpy($preffix$$name$, aux.c_str(), aux.size() + 1);
+}
+$else$
+dcdr \>> $preffix$$name$;
+$endif$
+$elseif(typecode.isStringType)$
+$if(typecode.isBounded)$
+{
+    std::string aux;
+    dcdr \>> aux;
+    $preffix$$name$ = aux.c_str();
+}
+$else$
+dcdr \>> $preffix$$name$;
+$endif$
+$else$
+dcdr \>> $preffix$$name$;
+$endif$
+>>
+
+cdrIndexName(name, loopvar) ::= <<$name$[$loopvar$]>>
+
+cdrMemberName(name, preffix) ::= <<$preffix$$name$>>
+
+member_default_init(ctx, member, loopvar) ::= <<
+// $member.typecode.idlTypename$ m_$member.name$
+$if(member.typecode.primitive)$$if(member.typecode.isBitmaskType)$m_$member.name$ = $if(member.annotationDefault)$$member.annotationDefaultValue$$else$static_cast<$member.typecode.cppTypename$>(0)$endif$;$else$m_$member.name$ = $if(member.annotationDefault)$$member.annotationDefaultValue$$else$$member.typecode.initialValue$$endif$;$endif$$elseif(member.typecode.isType_f)$$if(member.typecode.contentTypeCode.primitive)$memset(&m_$member.name$, $if(member.annotationDefault)$$member.annotationDefaultValue$$else$0$endif$, $member.typecode.size$ * $member.typecode.contentTypeCode.size$);$elseif(member.typecode.contentTypeCode.isStringType)$$if(ctx.generateTypesC)$$member_array_default_cstring_init(ctx=ctx, name=memberName(member.name), loopvar=ctx.newLoopVarName, dims=member.typecode.dimensions)$$endif$$endif$$endif$>>
+
+member_array_default_cstring_init(ctx, name, loopvar, dims) ::= <<$if(rest(dims))$for (uint32_t $loopvar$ = 0; $loopvar$ < $name$.size(); ++$loopvar$)
+{
+    $member_array_default_cstring_init(ctx=ctx, name=indexName(name=name,loopvar=loopvar),  loopvar=ctx.nextLoopVarName, dims=rest(dims))$
+}$else$$name$.fill(nullptr);$endif$>>
+
+indexName(name, loopvar) ::= <<$name$[$loopvar$]>>
+
+memberName(name) ::= <<m_$name$>>
+
+member_destructor(ctx, member) ::= <<$if(ctx.generateTypesC)$$if(member.typecode.isStringType)$free(m_$member.name$);
+m_$member.name$ = nullptr;$elseif(member.typecode.isType_f)$$if(member.typecode.contentTypeCode.isStringType)$$member_array_cstring_destructor(ctx=ctx, name=memberName(member.name), loopvar=ctx.newLoopVarName, dims=member.typecode.dimensions)$$endif$$endif$$endif$>>
+
+member_array_cstring_destructor(ctx, name, loopvar, dims) ::= <<$if(rest(dims))$for (uint32_t $loopvar$ = 0; $loopvar$ < $name$.size(); ++$loopvar$)
+{
+    $member_array_cstring_destructor(ctx=ctx, name=indexName(name=name,loopvar=loopvar),  loopvar=ctx.nextLoopVarName, dims=rest(dims))$
+}$else$for (char* str : $name$)
+{
+    free(str);
+}$endif$>>
```

Comment:
Several of these methods are already defined in FastCdrCommon.stg which is included in this file. I suppose they can be safely removed.

Reply:
Done in 020947b